pkg/controller/vsphere/session/session.go

```diff
@@ -105,20 +105,20 @@ func GetOrCreate(
 	return &session, nil
 }
 
-func (s *Session) FindVM(ctx context.Context, ID string) (*object.VirtualMachine, error) {
-	if !isValidUUID(ID) {
-		klog.V(3).Infof("Find template by instance uuid %v", ID)
-		ref, err := s.FindRefByInstanceUUID(ctx, ID)
-		if err != nil {
-			return nil, errors.Wrap(err, "error querying template by instance UUID")
-		}
-		if ref != nil {
-			return object.NewVirtualMachine(s.Client.Client, ref.Reference()), nil
-		}
+func (s *Session) FindVM(ctx context.Context, UUID, name string) (*object.VirtualMachine, error) {
+	if !isValidUUID(UUID) {
+		klog.V(3).Infof("No valid UUID %v, trying to find by name", name)
```

Comment:
Should this log message include the UUID given? 
```suggestion
		klog.V(3).Infof("Invalid UUID for VM %q: %s, trying to find by name", name, UUID)
```